src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -169,13 +169,13 @@ class ZooKeeperSink : public SinkToStorage
 /// Type of path to be fetched
 enum class ZkPathType
 {
-    Exact,   /// Fetch all nodes under this path
-    Prefix,  /// Fetch all nodes starting with this prefix, recursively (multiple paths may match prefix)
-    Recurse, /// Fatch all nodes under this path, recursively
+    Exact, /// Fetch all nodes under this path
+    Prefix, /// Fetch all nodes starting with this prefix, recursively (multiple paths may match prefix)
+    Recurse, /// Fetch all nodes under this path, recursively
 };
 
-/// List of paths to be feched from zookeeper
-using Paths = std::deque<std::pair<String, ZkPathType>>;
+/// List of paths to be fetched from zookeeper
+using Paths = std::unordered_map<String, ZkPathType>;
```

Comment:
Use an unordered_map to deduplicate when possible before making requests to keeper